x-pack/legacy/plugins/apm/server/lib/settings/agent_configuration/get_environments/get_existing_environments_for_service.ts

```diff
@@ -42,7 +44,10 @@ export async function getExistingEnvironmentsForService({
     }
   };
 
-  const resp = await client.search(params);
+  const resp = (await getCallWithInternalUserFromSetup(setup)(
+    'search',
+    params
+  )) as ESSearchResponse<unknown, typeof params>;
```

Comment:
I find this a little hard to read. I don't think the optional introduced by `allowCallWithInternalUser` is super important. 

What do you think of something like:

```ts
const { callWithInternalUser } = setup;
const resp = await callWithInternalUser(params);
```

`allowCallWithInternalUser` could then be defined in a similar fashion to `getESClient` where the types for each operation are already defined.

Reply:
I was trying to make it explicit but I can see that it's hard to read. I do think we should express in some way that this is an exception though, especially because the call itself is far removed from the endpoint protection. Always exposing `callWithInternalUser` on `Setup` makes it hard to see what endpoints are using it. Do you think passing in `callWithInternalUser` as an argument to the function is a good compromise? 

I originally opted to not use something like `getESClient` due to time constraints but I can see the value of having a similar approach for `callWithInternalUser`.

Reply:
> Do you think passing in callWithInternalUser as an argument to the function is a good compromise?

Yes, sounds like a good compromise 👍

Reply:
Per our discussion on Slack, changed this implementation to let whoever calls `setupRequest` decide whether the client used for ES should use `callWithInternalUser` or `callWithRequest`.